### PR TITLE
docs: Redrafts readme to capture vision

### PR DESCRIPTION
**Pull request by Frijol**
My friends! I watched/read the latest Archiving meeting and think I have a better idea of what we're trying to do. I've taken my understanding and drafted it into readme form.

## Please review!
The primary purpose of this is to check that we're all on the same page. **It's pretty likely I got something wrong, so please read carefully and poke holes! That's what a straw-type draft is for.**

Secondary purpose (of course) is to document what we're doing for better communication internally & externally– suggestions welcome here as well.

---
Other questions:
* In this PR, I deleted basically all of the old readme content because it looked like somebody's personal notes out of context. Are these still useful/is there a place I should put them?
* There's some overlap and some informational conflict with roadmap.md. I wasn't sure what to do with that document. What is useful from it? What is out of date? Can I get rid of it and put the useful parts in the readme?

Overall, this PR and the original roadmap.md feel more like RFC process than straight content revision, but I think we can work with that in PR form..

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,87 +1,62 @@
...
+The crawler crawls just one level (links from one particular page) at a time. The results are then sent to the coordinator/result handler, which performs some minimal processing (e.g. de-duplication of lists of returned URLs) and queues the next iterations of crawling.
 
-Thing rob had started writing:
+![Very small/blurry diagram of the jobs, workers, and results interacting with the coordinator](https://lh3.googleusercontent.com/W2HUfQXgmz-MQQfc7Z1QpeAo1dS7WqAZ8AziG-uYYVZu_GRCjXDIhgIuqcV9WVp8cFMtV-IAjWsRMqohxSWPPHowLXZritm8QHqwLiAltDBxqodZX54Ogo3mkOmYCUKgGADN5hjx8RvCbh8-adJfqjRZkmCFjNgVRFha_5V-pa9PiSTAGkB3I2GphzQT9qJ4GDgplogezKWYX43kWzszcsofEHhkow4LPvV7xVYHh3HSo-AbKtKdD8q_ZAqO2gmn2h_2HXv8SPqqjH2Rbft3GwHcIPrSaFqInwhryzSakkHLrDA1R6AcC_9CxGh3t_fCvJSPmuxMK15dzuWnauoD2MlAcS_AlhOmtQFP7p-zW-OWpzqgMlYtnMYYrcXWefycnaf-Bv46sU_xGae_-ex3yk2TgPCmSwVVx8l3EK7aiSDYM0rqGCZS6w5SXYBlTVXnn4XYv2qToabUX0IDFlopuvMmzCiAg_gzNeou0_Q_8Zx56cK2yMnJkJp9G_oHu-ndQ2hJRTD4dcV0LcjQoFwZ86jurcTcjc0mpv-4caHgxDEzXQVgzscCwhSvRLHGcjxZNghOSFknXSOIqCmVVB_c9EQv3K4scHXliJkYfyr00CRAYG-UeoDWhylWTETMpktWnlt0VaOTTrn6sVfuueMFIBGGxE6jY_wq=w317-h99-no)
```
Comment (lightandluck) [inline readme.md]: This image is broken for me.
When I try to click on it while viewing file it takes me to this link: https://camo.githubusercontent.com/2f0c9beed9666b7c970053b8ddc57bd0a998fe17/68747470733a2f2f6c68332e676f6f676c6575736572636f6e74656e742e636f6d2f57324855665158676d7a2d4d51516663375a31517065416f316453375771415a38417a69472d755959565a755f4752436a58444968674975716356395756703863464d74562d49416a5773524d716f687853575050486f774c585a7269746d38514871774c69416c74444278716f645a5835344f676f336d6b4f6d5943554b674741444e35686a78385276436268382d61644a66716a525a6b6d43466a4e6756524668615f35562d7061395069535441476b423349324770687a515439714a34474467706c6f67657a4b57595834336b577a737a63736f664548686b6f77344c5076563778565948683348536f2d41624b744b644438715f5a41714f32676d6e32685f3248587638535071716a483252626674334777486349507253614671496e776872797a53616b6b484c7244413152364163435f394378476833745f6643764a53506d75784d4b3135647a75576e61756f44324d6c4163535f416c684f6d7451465037702d7a572d4f57707a71674d6c59746e4d595972635857656679636e61662d4276343673555f784761655f2d657833796b32546750436d5377565678386c33454b3761695344594d30727147435a53367735535859426c5456586e6e345859763271546f61625558304944466c6f7075764d6d7a436941675f677a4e656f75305f515f385a783536634b32794d6e4a6b4a7039475f6f48752d6e645132684a52544434646356304c636a516f46775a38366a75726354636a63306d70762d34636148677844457a585156677a73634377685376524c4847636a785a4e67684f53466b6e58534f4971436d5656425f6339455176334b34736348586c694a6b59667972303043524159472d55656f445768796c575445544d706b74576e6c743056614f5454726e367356667575654d46494247477845366a595f77713d773331372d6839392d6e6f

I tried to go directly to `lh3.googleusercontent.com/...` instead, but that's broken as well.

**Comment by b5** [comment]
Eep, while **I'm not sure we have consensus on this change**, you shouldn't have to screen-grab. Here's a proper diagram:

Artboard 1

File: readme.md (hunk 1 above)
Comment (Frijol) [reply readme.md]: Oh I worried it might do that. Ha @b5's is a better solution

Comment (lightandluck) [reply readme.md]: 👍 Works now!

**Review by danielballan — APPROVED** [review]
This hits all the points it needs to hit, from my point of view. Thanks @Frijol!

**Pull request merged**